roofit/xroofit/src/xRooFit.cxx

```diff
@@ -340,15 +362,15 @@ xRooFit::generateFrom(RooAbsPdf &pdf, const std::shared_ptr<const RooFitResult>
          _out.first->add(_tmp);
       } else {
          if (_pdf->canBeExtended()) {
-            _out.first = std::unique_ptr<RooDataSet>{_pdf->generate(*_obs, RooFit::Extended(), RooFit::ExpectedData(expected))};
+            _out.first.reset(_pdf->generate(*_obs, RooFit::Extended(), RooFit::ExpectedData(expected)));
```

Comment:
```suggestion
            _out.first = std::unique_ptr<RooDataSet>{_pdf->generate(*_obs, RooFit::Extended(), RooFit::ExpectedData(expected))};
```

Please revert that change and use `std::unique_ptr`.

The reason is that I regularly build RooFit with this "memory safe" mode, where all owning pointers are unique pointers:
https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooFit/Config.h#L20

It's a very useful tool to check for memory leaks. It would therefore be important that also xroofit still complies when this `ROOFIT_MEMORY_SAFE_INTERFACES` flag is defined in the RooFit config header.